File: src/geocaching/api/impl/live_geocaching_api/filter/TerrainFilter.java
package geocaching.api.impl.live_geocaching_api.filter;

import google.gson.stream.JsonWriter;

import java.io.IOException;

public class TerrainFilter implements Filter {
	private static final String NAME = "Terrain";
	
	protected final float min;
	protected final float max;
	
	public TerrainFilter(float min, float max) {
		this.min = min;
		this.max = max;
	}
	
	@Override
	public String getName() {
		return NAME;
	}
	
	public float getMin() {
		return min;
	}
	
	public float getMax() {
		return max;
	}

	@Override
	public boolean isValid() {
		return min != 1 && max != 5;
	}
	
	@Override
	public void writeJson(JsonWriter w) throws IOException {
		w.name(NAME);
		w.beginObject();
		
		w.name("MinTerrain").value(min);
		w.name("MaxTerrain").value(max);
		
		w.endObject();
	}
}

File: geocaching-api/src/main/java/com/arcao/geocaching4locus/data/api/model/request/query/filter/TerrainFilter.kt
package com.arcao.geocaching4locus.data.api.model.request.query.filter

class TerrainFilter(private val min: Float, private val max: Float) : Filter {
    override fun isValid() = min >= 1F && max <= 5F && min <= max

    override fun toString(): String {
        return "terr:$min-$max"
    }
}
